system/settings/settings.xml

```diff
@@ -1710,17 +1710,19 @@
           </updates>
           <control type="edit" format="integer" />
         </setting>
+        <setting id="services.webserverauthentication" parent="services.webserver" type="boolean" label="1052" help="1053">
+          <level>2</level>
+          <default>true</default>
+          <control type="toggle" />
+        </setting>
         <setting id="services.webserverusername" type="string" parent="services.webserver" label="1048" help="36330">
           <level>1</level>
           <default>kodi</default>
-          <constraints>
```

Comment:
According to my tests, if the stored value is empty, it will be reset to the default value (`kodi`) automatically and without user interaction. So there is no need for extra upgrade processing.

By the way, this change does break installations which did not have a username, but a password, set before (which is technically valid for HTTP authentication, I guess?). Anyone think this is an issue?

Reply:
I didn't even know this was valid.

Reply:
Let's ignore that particular problem for now then